src/game/state.cc

```diff
@@ -0,0 +1,68 @@
+#include <ctime>
+
+#include "state.h"
+
+namespace SailGame { namespace Game {
+
+GameState::GameState(const std::vector<unsigned int> &userIds, 
+    const StartGameSettings &gameSettings)
+    : mPlayerNum(userIds.size()),
+    mGameSettings(std::make_unique<StartGameSettings>(gameSettings))
+{
+    for (auto userId : userIds) {
+        mUserIdToPlayerState.emplace(userId, PlayerState{});
+    }
+
+    mDiscardPile = std::make_unique<DiscardPile>();
```

Comment:
don't see the benefit of unique_pointer.

Reply:
frankly speaking, I was worried to mix up the order to construct `DiscardPile` and `Deck`. anyway, change to object now